### PR TITLE
Eislam/sy 1130 refactor opc reader

### PR DESCRIPTION
**Pull request by Lham42**
# Fix Pull Request Template

## Key Information

- **Linear Issue**: [SY-1130]()

## Description

This pr refactors code for the opc reader so that headers are in `reader.h` and implementations are separated into `reader_source.cpp` and `reader_task.cpp`. This pr should not be reviewed/merged until the opc [ua write support](https://github.com/synnaxlabs/synnax/pull/792) is merged in 

## Basic Readiness Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have added sufficient regression tests to cover the changes to CI.
- [ ] I have added relevant tests to cover the changes or exposing bugs.
- [ ] I have verified code coverage targets are met.

## Additional Notes
- [ ] These changes deal with concurrency.
- [ ] These changes affect UI.

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
  with necessary manual QA steps to test my change.

## Reviewer Checklist
- [ ] Sufficient test coverage of new additions.
- [ ] Verified all steps in readiness checklists.
- [ ] UI changes have been tested.
- [ ] style and formatting is consistent.
- [ ] Reviewed any relevant changes to concurrent code for safety. 
- [ ] Sufficient comments and clarity of code.

### CODE REVIEW

File: driver/opc/reader_task.cpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        ctx->setState({
+            .task = task.key,
+            .variant = "error",
+            .details = json{{"message", err.message()}}
+        });
+        return nullptr;
+    }
+    auto [channel_keys, indexes] = res;
+
+    // Connect to the OPC UA server.
```
Comment (emilbon99) [inline driver/opc/reader_task.cpp:134]: Useless comment

File: driver/opc/reader_task.cpp [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+//                                    Reader Task                                //
+///////////////////////////////////////////////////////////////////////////////////
+std::unique_ptr<task::Task> ReaderTask::configure(
+    const std::shared_ptr<task::Context> &ctx,
+    const synnax::Task &task
+) {
+    VLOG(2) << "[opc.reader] configuring task " << task.name;
+    auto config_parser = config::Parser(task.config);
+    auto cfg = ReaderConfig(config_parser);
+    //    LOG(INFO) << "Reader Config: " << config_parser.get_json().dump(4);
```
Comment (emilbon99) [inline driver/opc/reader_task.cpp:89]: Should we remove this?

File: driver/opc/reader_task.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        breaker_config,
+        std::move(source),
+        writer_cfg,
+        ua_client,
+        properties
+    );
+}
+
+void ReaderTask::exec(task::Command &cmd) {
+    if (cmd.type == "start") {
```
Comment (emilbon99) [inline driver/opc/reader_task.cpp:218]: No need for braces

File: driver/opc/util.cpp [hunk 4]
```diff
@@ -291,8 +291,8 @@ std::pair<std::shared_ptr<UA_Client>, freighter::Error> opc::connect(
     config->timeout = 7200000; // (ms) 2 hours
     configure_encryption(cfg, client);
     UA_StatusCode status;
-//    freighter::Error err = set_authentication(cfg, config);
-//    if(!err) return {std::move(client), err};
+    //    freighter::Error err = set_authentication(cfg, config);
```
Comment (emilbon99) [inline driver/opc/util.cpp:294]: Can we remove this?

**Review by emilbon99 — APPROVED** [review]
Approved once you address minor comments

File: driver/opc/reader_task.cpp (hunk 2 above)
Comment (Lham42) [reply driver/opc/reader_task.cpp:89]: done

**Pull request merged**